internal/auth/header_builder.go

```diff
@@ -0,0 +1,115 @@
+package auth
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	ec "github.com/bitcoin-sv/go-sdk/primitives/ec"
+	"github.com/bitcoin-sv/spv-wallet/models"
+)
+
+// HeaderConfig contains configuration settings for request header authentication.
+// It defines options for signing requests and selecting the appropriate key for signing.
+type HeaderConfig struct {
+	SignRequest bool               // SignRequest indicates whether to sign the request.
+	PrivateKey  *ec.PrivateKey     // PrivateKey is used for signing when no ExtendedKey is provided.
+	ExtendedKey *bip32.ExtendedKey // ExtendedKey is used for signing when provided.
```

Comment:
It is misleading (or at least I was confised when I was reading it).
Consider more strait-forward approach:
```suggestion
	AccessKey *ec.PrivateKey  //some desc (IF NECESSARY) 
	XPriv *bip32.ExtendedKey 
	XPub *bip32.ExtendedKey 
```

And it would simplify decision when to `sign`

```golang
func (h HeaderConfig) IsSignRequest() bool { 
    return h.XPriv != nil || h.AccessKey != nil
}
```